lib/cylc/network/https/base_client.py

```diff
@@ -48,27 +54,73 @@ class BaseCommsClient(object):
     METHOD_GET = 'GET'
 
     def __init__(self, suite, owner=USER, host=None, port=None, timeout=None,
-                 my_uuid=None, print_uuid=False):
+                 my_uuid=None, print_uuid=False, comms_protocol=None):
         self.suite = suite
         self.owner = owner
         self.host = host
         self.port = port
+        self.srv_files_mgr = SuiteSrvFilesManager()
+        self.comms_protocol = comms_protocol
         if timeout is not None:
             timeout = float(timeout)
         self.timeout = timeout
         self.my_uuid = my_uuid or uuid4()
         if print_uuid:
             print >> sys.stderr, '%s' % self.my_uuid
-        self.srv_files_mgr = SuiteSrvFilesManager()
+
         self.prog_name = os.path.basename(sys.argv[0])
         self.server_cert = None
         self.auth = None
 
-    def _compile_url(self, category, func_dict, host):
+    def _get_comms_from_suite_contact_file(self):
+        """Find out the communications protocol (http/https) from the
+        suite contact file."""
+        try:
+            comms_prtcl = self.srv_files_mgr.get_auth_item(
+                self.srv_files_mgr.KEY_COMMS_PROTOCOL,
+                self.suite, content=True)
+            if comms_prtcl is None or comms_prtcl == "":
+                raise TypeError("Comms protocol is not in suite contact file")
+            else:
+                return comms_prtcl
+        except (AttributeError, KeyError, TypeError, ValueError):
+            raise KeyError("No suite contact info for comms protocol found")
+
```

Comment:
It will fall back to finding it from the global.rc file if not in the contact file here, which already has a default setting of https in previous versions. But yes, what if the old running suite is using http...I will have to think about how to handle this...